core/src/main/java/org/apache/brooklyn/core/mgmt/BrooklynTags.java

```diff
@@ -108,6 +123,39 @@ public TraitsTag(List<Class<?>> interfaces) {
         }
     }
 
+    public static class CampBrooklynTag implements Serializable {
```

Comment:
I'm not convinced we want to wrap the user's tag in a `CampBrooklynTag`, rather than just adding their tag directly. I'm guessing that things behave very differently if you do `brooklyn.tags: ["hi"]` then the result is very different from doing Java `entity.tags()addTag("hi")` - i.e. the latter has the tag "hi", whereas the former has the tag of type "CampBrooklynTag" that wraps the string "hi".

Other than being able to tell where the tag came from, what is the advantage of using `CampBrooklynTag`?